models/DataObject/SelectOptions/Data/SelectOption.php

```diff
@@ -0,0 +1,108 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\DataObject\SelectOptions\Data;
+
+class SelectOption implements \JsonSerializable
+{
+    public const PROPERTY_VALUE = 'value';
+    public const PROPERTY_LABEL = 'label';
+    public const PROPERTY_NAME = 'name';
+
+    public function __construct(
+        protected string $value,
+        protected string $label,
+        protected string $name = '',
+    ) {}
+
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setValue(string $value): static
+    {
+        $this->value = $value;
+        return $this;
+    }
+
+    public function hasValue(): bool
+    {
+        return !empty($this->value);
+    }
+
+    public function getLabel(): string
+    {
+        return $this->label;
+    }
+
+    public function hasLabel(): bool
+    {
+        return !empty($this->label) && $this->label !== '0';
```

Comment:
Since it's a `string`, you could just write (also for the other fields):

```suggestion
        return $this->label !== '';
```